app/src/main/java/org/mozilla/fenix/onboarding/DefaultBrowserNotificationWorker.kt

```diff
@@ -21,66 +25,37 @@ import mozilla.components.support.base.ids.SharedIdsHelper
 import org.mozilla.fenix.GleanMetrics.Events
 import org.mozilla.fenix.HomeActivity
 import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.areNotificationsEnabledSafe
 import org.mozilla.fenix.ext.settings
 import org.mozilla.fenix.utils.IntentUtils
 import org.mozilla.fenix.utils.Settings
 import java.util.concurrent.TimeUnit
 
 class DefaultBrowserNotificationWorker(
-    context: Context,
+    private val context: Context,
     workerParameters: WorkerParameters,
 ) : Worker(context, workerParameters) {
 
     override fun doWork(): Result {
         val channelId = ensureMarketingChannelExists(applicationContext)
 
-        NotificationManagerCompat.from(applicationContext)
-            .notify(
-                NOTIFICATION_TAG,
-                DEFAULT_BROWSER_NOTIFICATION_ID,
-                buildNotification(channelId),
-            )
-
-        Events.defaultBrowserNotifShown.record(NoExtras())
-
-        // default browser notification should only happen once
-        applicationContext.settings().defaultBrowserNotificationDisplayed = true
-
-        return Result.success()
-    }
-
-    /**
-     * Build the default browser notification.
-     */
-    private fun buildNotification(channelId: String): Notification {
-        val intent = Intent(applicationContext, HomeActivity::class.java)
-        intent.putExtra(INTENT_DEFAULT_BROWSER_NOTIFICATION, true)
-
-        val pendingIntent = PendingIntent.getActivity(
-            applicationContext,
-            SharedIdsHelper.getNextIdForTag(applicationContext, NOTIFICATION_PENDING_INTENT_TAG),
-            intent,
-            IntentUtils.defaultIntentPendingFlags,
-        )
-
-        with(applicationContext) {
-            val appName = getString(R.string.app_name)
-            return NotificationCompat.Builder(this, channelId)
-                .setSmallIcon(R.drawable.ic_status_logo)
-                .setContentTitle(
-                    applicationContext.getString(R.string.notification_default_browser_title, appName),
-                )
-                .setContentText(
-                    applicationContext.getString(R.string.notification_default_browser_text, appName),
+        val notificationManagerCompat = NotificationManagerCompat.from(applicationContext)
+
+        @SuppressLint("MissingPermission") // notification permission is checked here
+        if (notificationManagerCompat.areNotificationsEnabledSafe()) {
+            showDefaultBrowserNotification(applicationContext, channelId)
+        } else {
+            // we can only request notification permission on Android 13 and above
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                requestPermissions(
+                    context as Activity,
```

Comment:
⚠️ This shouldn't be cast to activity, this will be the application that will trigger the worker to run in the background. 

We are requesting permission as part of the work here, should we request permission while setting up the work and just build notification if the permission is granted here as part of `doWork`.

Reply:
Regarding the cast, this is not the final code for this part, I just put this together to demonstrate the intention.

As for the moment we request the permission, there is no guarantee that the state of the permission will remain the same from the moment we set up the work to the moment we show the notification.